pkg/apis/network/v1/egressrouter_types.go

```diff
@@ -0,0 +1,121 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EgressRouter is a feature allowing the user to define an egress router
+// that acts as a bridge between pods and external systems. The egress router runs
+// a service that redirects egress traffic originating from a pod or a group of
+// pods to a remote external system or multiple destinations as per configuration.
+
+//  More specifically, given an EgressRouter CR with <name>, the CNO will create and manage:
+// - A service called <name>
+// - An egress pod called <name>
+// - An NAD called <name>
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// EgressRouter is a single egressrouter pod configuration object.
+// +k8s:openapi-gen=true
+// +kubebuilder:resource:path=egressrouters,scope=Namespaced
+// +kubebuilder:resource:shortName=egr
+// +kubebuilder:printcolumn:name="EgressRouters",type=string,JSONPath=".status.egressRouter"
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=".status.status"
+type EgressRouter struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	Spec EgressRouterSpec `json:"spec"`
+
+	// +kubebuilder:validation:Optional
+	// +optional
+	Status EgressRouterStatus `json:"status,omitempty"`
+}
+
+// EgressRouterSpec contains the configuration for an egress router.
+// +k8s:openapi-gen=true
+// +kubebuilder:validation:Required
+type EgressRouterSpec struct {
+	// Specification of interface to create/use. e.g; macvlan
+	// +kubebuilder:validation:Required
+	NetworkInterface EgressRouterInterface `json:"networkInterface"`
+
+	// IP configuration arguments.
+	// +kubebuilder:validation:Required
+	IP IPConfig `json:"ip"`
+}
+
+// EgressRouterInterface contains the configuration of interface to create/use.
+// +kubebuilder:validation:Required
+type EgressRouterInterface struct {
+	// Arguments specific to the interfaceType macvlan
+	// +kubebuilder:validation:Required
+	MacVLAN VLANConfig `json:"macVLAN"`
+}
+
+// VLANConfig consists of arguments specific to the EgressRouterInterfaceType
+// +kubebuilder:validation:Required
+type VLANConfig struct {
+	// Usually used for macvlan interface.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Optional
+	// +optional
+	Mode string `json:"mode,omitempty"`
```

Comment:
Again the mode isn't a random string, so it should be a type, the deploymentConfig is a good example:

https://github.com/openshift/api/blob/992bc3a41fe6c4e5bcf27dc8080f9e5c114c33fc/apps/v1/types.go#L110-L119